src/modules/apps/e2e/index.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react'
+import { render } from 'ink'
+
+import { apps, tester } from '../../../clients'
+import { RealTimeReport } from './report'
+import { getToken } from '../../../conf'
+import { ManifestEditor } from '../../../lib/manifest/ManifestEditor'
+import { ErrorBoundary } from './errorBoundary'
+import { TestRequest } from '../../../clients/Tester'
+
+
+export default async options => {
+  new EndToEndCommand(options).run()
+}
+
+class EndToEndCommand {
+  options : any
+
+  constructor(options) {
+    this.options = options
+  }
+  public run() {
+    if(this.options.workspace) {
+      return this.runWorkspaceTests()
+    }
+
+    return this.runAppTests()
+  }
+
+  private async runAppTests() {
+    const manifestEditor = await ManifestEditor.getManifestEditor()
+    const cleanAppId = manifestEditor.appLocator
+
+    const appList = await apps.listApps()
+    const appId = appList.data.map(({ app }) => app).find(app => app.startsWith(cleanAppId))
```

Comment:
_nit_: How about removing this `map`?
```suggestion
    const appId = appList.data.find(({ app }) => app.startsWith(cleanAppId))
```

Reply:
I think this `find` will not work in case `cleanAppId = vtex.asdf@0.x` and the workspace has `vtex.asdf@1.x` installed - it will run the tests anyway:

The list returned from apps has the format:
```
   {
      app: 'vtex.ab-tester@0.16.2',
      id: 'vtex.ab-tester@0.16.2',
      _source: 'installation',
      location: '/vtex/n1poli/apps/vtex.ab-tester@0.16.2'
    },
    {
      app: 'vtex.admin@3.7.0',
      id: 'vtex.admin@3.7.0',
      _source: 'installation',
      location: '/vtex/n1poli/apps/vtex.admin@3.7.0'
    },
    {
      app: 'vtex.admin-graphql-ide@3.1.0',
      id: 'vtex.admin-graphql-ide@3.1.0',
      _source: 'installation',
      location: '/vtex/n1poli/apps/vtex.admin-graphql-ide@3.1.0'
    },
```
So `startsWith` is not enough, may will have to use `@vtex/api`'s `formatMajorLocator` and then compare the two strings

Reply:
Sorry, had a brain fart - I think it's everything ok with this